backend/spec/requests/api/v1/reset_password_spec.rb

```diff
@@ -0,0 +1,141 @@
+require "swagger_helper"
+
+RSpec.describe "API V1 Reset Password", type: :request do
+  let!(:user) { create(:user, email: "user@example.com", password: "SuperSecret1234") }
+
+  path "/api/v1/reset_password" do
+    post "Sends an email with reset password path and token" do
+      tags "Reset Password"
+      consumes "application/json"
+      produces "application/json"
+      security [csrf: []]
+      parameter name: :params, in: :body, schema: {
+        type: :object,
+        properties: {
+          email: {type: :string}
+        },
+        required: ["email"]
+      }
+
+      response "200", :success do
+        let(:params) { {email: "user@example.com"} }
+        let("X-CSRF-TOKEN") { get_csrf_token }
+
+        run_test!
+
+        it "sends email" do
+          mail = ActionMailer::Base.deliveries.last
+          expect(mail.subject).to eq("Reset password instructions")
+          expect(mail.to.first).to eq("user@example.com")
+          expect(user.reload.reset_password_token).to be_present
+        end
+
+        context "invalid email" do
+          let(:params) { {email: "invalid"} }
+
+          it "returns 200" do
+            expect(response).to have_http_status(:ok)
+          end
+        end
+
+        context "non existing user" do
+          let(:params) { {email: "valid@example.com"} }
+
+          it "returns 200" do
+            expect(response).to have_http_status(:ok)
+          end
+        end
+
+        context "wrong params" do
+          let(:params) { {} }
+
+          it "returns 200" do
+            expect(response).to have_http_status(:ok)
+          end
+        end
+      end
+    end
+
+    put "Resets password" do
+      tags "Reset Password"
+      consumes "application/json"
+      produces "application/json"
+      security [csrf: []]
+      parameter name: :params, in: :body, schema: {
+        type: :object,
+        properties: {
+          password: {type: :string},
+          reset_password_token: {type: :string}
+        },
+        required: ["email", "reset_password_token"]
+      }
+
+      response "200", :success do
+        schema type: :object, properties: {
+          data: {"$ref" => "#/components/schemas/user"}
+        }
+        let(:token) {
+          user.send(:set_reset_password_token) # this is protected method
```

Comment:
maybe there is some magic happening inside this method, but is it not more readable when use set token manually?
```
create :user, reset_password_token: token, reset_password_sent_at: Time.current
```

Reply:
Not sure if that will work, reset_password_token is different than confirmation_token (dunno why, some devise inconsistency). It is encrypted in db, but I can check if it works.

Reply:
I think I will leave that protected method. Otherwise I have to generate by hand like this https://mokacoding.com/blog/how-to-manually-generate-devise-reset-password-link/ and save hashed in db, and send raw as param.